### PR TITLE
handle each potentially dangerous command

### LINKED ISSUE #676: Fix 'destroy infrastructure' in CI pipeline
destroy infrastructure step failed in CI pipeline:
https://buildkite.com/cyberway/neon-proxy/builds/5224#40be2d24-7a76-4a1b-bb31-f1340e50397b

### PR DESCRIPTION
**Pull request by inkognitotester**

### CODE REVIEW

**Review by vodolaz — APPROVED** [review]

File: .buildkite/steps/full_test_suite/terraform-destroy.sh [hunk 1]
```diff
@@ -1,5 +1,15 @@
 #!/bin/bash
-set -euo pipefail
+
+# error handler
```
Comment (rozhkovdmitrii) [inline .buildkite/steps/full_test_suite/terraform-destroy.sh]: if you want to make the code more expressive just name the function `handle_error`. Please get rid of comments.

File: .buildkite/steps/full_test_suite/terraform-destroy.sh [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+check_res() {
+  if [[ $? -ne 0 ]]
+  then
+    # Print error message and interrupt script
```
Comment (rozhkovdmitrii) [inline .buildkite/steps/full_test_suite/terraform-destroy.sh]: the same -  just get rid of this line

File: .buildkite/steps/full_test_suite/terraform-destroy.sh [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+    echo "Interrupt at step: $1"
```
Comment (rozhkovdmitrii) [inline .buildkite/steps/full_test_suite/terraform-destroy.sh]: echo "Interrupt at step. $1"

File: .buildkite/steps/full_test_suite/terraform-destroy.sh [hunk 4]
```diff
@@ -8,13 +18,16 @@ cd .buildkite/steps/full_test_suite
 export SSH_KEY="~/.ssh/ci-stands"
 export ARTIFACTS_LOGS="./logs"
 mkdir -p $ARTIFACTS_LOGS
+check_res "Create directory for logs '$ARTIFACTS_LOGS'"
```
Comment (rozhkovdmitrii) [inline .buildkite/steps/full_test_suite/terraform-destroy.sh]: check_res "Failed to create artifacts dir at: '$ARTIFACTS_LOGS'"

File: .buildkite/steps/full_test_suite/terraform-destroy.sh [hunk 5]
```diff
@@ -8,13 +18,16 @@ cd .buildkite/steps/full_test_suite
...
 
 
 # solana
 export REMOTE_HOST=`buildkite-agent meta-data get "SOLANA_IP"`
 ssh-keyscan -H $REMOTE_HOST >> ~/.ssh/known_hosts
 ssh -i ${SSH_KEY} ubuntu@${REMOTE_HOST} 'sudo docker logs solana > /tmp/solana.log 2>&1'
+check_res "Dump solana logs to the file"
```
Comment (rozhkovdmitrii) [inline .buildkite/steps/full_test_suite/terraform-destroy.sh]: Failed to dump logs remotely at: ' ubuntu@${REMOTE_HOST}' with key file: '${SSH_KEY}'

File: .buildkite/steps/full_test_suite/terraform-destroy.sh [hunk 6]
```diff
@@ -8,13 +18,16 @@ cd .buildkite/steps/full_test_suite
 export SSH_KEY="~/.ssh/ci-stands"
 export ARTIFACTS_LOGS="./logs"
 mkdir -p $ARTIFACTS_LOGS
+check_res "Create directory for logs '$ARTIFACTS_LOGS'"
 
 
 # solana
 export REMOTE_HOST=`buildkite-agent meta-data get "SOLANA_IP"`
 ssh-keyscan -H $REMOTE_HOST >> ~/.ssh/known_hosts
```
Comment (rozhkovdmitrii) [inline .buildkite/steps/full_test_suite/terraform-destroy.sh]: to be checked. Don't forget the context please

File: .buildkite/steps/full_test_suite/terraform-destroy.sh [hunk 7]
```diff
@@ -41,7 +56,9 @@ export TFSTATE_KEY="tests/test-$BUILDKITE_COMMIT"
 export TFSTATE_REGION="us-east-2"
 export TF_BACKEND_CONFIG="-backend-config="bucket=${TFSTATE_BUCKET}" -backend-config="key=${TFSTATE_KEY}" -backend-config="region=${TFSTATE_REGION}""
 terraform init $TF_BACKEND_CONFIG
+check_res "Terraform init step"
```
Comment (rozhkovdmitrii) [inline .buildkite/steps/full_test_suite/terraform-destroy.sh]: Failed to proceed the terraform init step. TF_BACKEND_CONFIG: ${TF_BACKEND_CONFIG}

File: .buildkite/steps/full_test_suite/terraform-destroy.sh [hunk 8]
```diff
@@ -41,7 +56,9 @@ export TFSTATE_KEY="tests/test-$BUILDKITE_COMMIT"
...
 terraform destroy --auto-approve=true
+check_res "Terraform destroy step"
```
Comment (rozhkovdmitrii) [inline .buildkite/steps/full_test_suite/terraform-destroy.sh]: Failed to proceed terraform destroy step

**Review by rozhkovdmitrii — CHANGES_REQUESTED** [review]

**Review by rozhkovdmitrii — APPROVED** [review]

**Pull request merged**